app/views/unattended/provisioning_templates/snippet/eol_puppet.conf.erb

```diff
@@ -0,0 +1,56 @@
+<%#
+kind: snippet
+name: eol_puppet.conf
+model: ProvisioningTemplate
+snippet: true
+description: |
+  legacy snippet in deprecation generates a puppet.conf file which is required for the puppet agent bootstraping.
+  The puppet server and CA is configured based on the host configuration. It supports Perforce Puppet legacy versionis.
+-%>
+<%
+  os_family = @host.operatingsystem.family
+  os_name   = @host.operatingsystem.name
+
+  aio_enabled = host_param_true?('enable-puppetlabs-repo') ||  host_param_true?('enable-official-puppet8-repo') || host_param_true?('enable-puppet8') || host_param_true?('enable-official-puppet7-repo') || host_param_true?('enable-puppet7') || host_param_true?('enable-puppetlabs-puppet6-repo') || host_param_true?('enable-puppet6') || host_param_true?('enable-puppetlabs-puppet5-repo') || host_param_true?('enable-puppet5')
+  aio_available = os_family == 'Debian' || os_family == 'Redhat' || os_name == 'SLES'
+
+  if os_family == 'Windows'
+    var_dir = 'C:\ProgramData\PuppetLabs\puppet\cache'
+    log_dir = 'C:\ProgramData\PuppetLabs\puppet\var\log'
+    run_dir = 'C:\ProgramData\PuppetLabs\puppet\var\run'
+    ssl_dir = 'C:\ProgramData\PuppetLabs\puppet\etc\ssl'
+  else
+    if @host.operatingsystem.family == 'Freebsd'
+      var_dir = '/var/puppet'
+    else
+      var_dir = '/var/lib/puppet'
+    end
+    log_dir = '/var/log/puppet'
+    run_dir = '/var/run/puppet'
+    ssl_dir = '\$vardir/ssl'
+  end
```

Comment:
It may be safe to drop these and rely on the defaults being present. At least Puppet AIO defaults to an empty `puppet.conf` since some time. So consider dropping them.